src/commands/open.js

```diff
@@ -1,25 +1,42 @@
 const log = require('@dhis2/cli-helpers-engine').reporter
-
+const concurrently = require('concurrently')
 const { cypress } = require('../tools/cypress.js')
+const { appStart, waitOn } = require('./common/sharedCLIOptions.js')
 
 exports.command = 'open'
 
 exports.aliases = ['o']
 
 exports.desc = 'Open Cypress UI'
 
-exports.builder = {}
+exports.builder = yargs =>
+    yargs.option('appStart', appStart).option('waitOn', waitOn)
 
 exports.handler = argv => {
     log.info('d2-utils-cypress > open')
-
-    const { port, browser } = argv
+    const { appStart, port, browser, waitOn } = argv
 
     const opts = {
         mode: 'open',
         browser,
         port,
     }
 
-    cypress(opts)
+    if (!appStart) {
+        log.info('"appStart" empty. Running cypress directly')
+
+        cypress(opts)
+    } else {
+        const cypressCommand = cypress({ ...opts, exec: false })
+        const waitOnCommand = `npx --no-install wait-on ${waitOn}`
+        const waitAndCypress = [waitOnCommand, cypressCommand].join(' && ')
+
+        concurrently(
+            [
+                { command: appStart, name: 'app' },
+                { command: waitAndCypress, name: 'cypress' },
+            ],
+            { kill: ['success'] }
+        )
+    }
```

Comment:
Very minor: could we factor this part out from this and the `run` command, since the code is identical?

Reply:
Should look much better now